components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1431,9 +1440,12 @@ void KeyringService::RemoveHardwareAccount(const std::string& address,
         GetPrefForKeyring(prefs_, kSelectedAccount, keyring_id);
     if (value && address == value->GetString()) {
       RemoveSelectedAccountForCoin(coin, keyring_id);
-      return;
     }
+    std::move(callback).Run(true);
+    return;
```

Comment:
It was doing unnecessary operations before when address is found but is not the selected account.